src/scripts/shop/ShopHandler.ts

```diff
@@ -37,12 +37,17 @@ class ShopHandler {
             } else {
                 Notifier.notify("You don't have enough money to buy " + this.amount() + " " + item.name() + multiple, GameConstants.NotificationOption.danger)
             }
-        } else if (item.currency == GameConstants.Currency.questpoint) {
+        } else if (item.currency == GameConstants.Currency.questpoint) { 
             if (player.hasQuestPoints(item.totalPrice())) {
                 player.payQuestPoints(item.totalPrice());
-                item.buy(this.amount());
-                item.increasePriceMultiplier(this.amount());
-                Notifier.notify("You bought " + this.amount() + " " + item.name() + multiple, GameConstants.NotificationOption.success)
+                if(item.name() =="Dungeon_ticket"){
```

Comment:
I'd rather have this in the buy method of key item i think

Reply:
did not know you can do that. built the keyitem.ts as a needed modification to allow it to be purchased - was not aware i can use the buy method. can you show me how once the bug is fixed?